cmd/profile/set.go

```diff
@@ -0,0 +1,75 @@
+package profile
+
+import (
+	"fmt"
+
+	"github.com/AlecAivazis/survey/v2"
+	"github.com/appgate/sdpctl/pkg/docs"
+	"github.com/appgate/sdpctl/pkg/profiles"
+	"github.com/appgate/sdpctl/pkg/prompt"
+	"github.com/spf13/cobra"
+)
+
+// NewSetCmd return a new profile set command
+func NewSetCmd(opts *commandOpts) *cobra.Command {
+	return &cobra.Command{
+		Use:     "set",
+		Short:   docs.ProfileSetDoc.Short,
+		Long:    docs.ProfileSetDoc.Long,
+		Example: docs.ProfileSetDoc.ExampleString(),
+		RunE: func(c *cobra.Command, args []string) error {
+			return setRun(c, args, opts)
+		},
+	}
+}
+
+func setRun(cmd *cobra.Command, args []string, opts *commandOpts) error {
+	if !profiles.FileExists() {
+		fmt.Fprintln(opts.Out, "no profiles added")
+		fmt.Fprintln(opts.Out, "run 'sdpctl profile add' first")
+		return nil
+	}
+	p, err := profiles.Read()
+	if err != nil {
+		return err
+	}
+	length := len(p.List)
```

Comment:
If I'm already using a default profile and create a new profile, once I set the new profile, I can't go back to my default profile again. It should be in the list of selectable profiles, but it's not added in the `profiles.json` file.

Reply:
I can't reproduce, or I don't understand. Could you elaborate?

Reply:
Ah, there seem to some wierdness here. Here's how to reproduce if you start with only having the `$SDPCTL_CONFIG_DIR/config.json` from before:
```bash
# initialise by trying to add a new profile called 'default'
> sdpctl profile add default
1 error occurred:
	* profile already exists with the name default

# list profiles and see that it's empty
> sdpctl profile ls
no current profile is set, run 'sdpctl profile set'

Available profiles
Name    Config directory
----    ----------------

# add a second profile
> sdpctl profile add secondary
Created profile secondary, run 'sdpctl profile list' to see all available profiles
run 'sdpctl profile set secondary' to select the new profile profile

# run ls again, still no default profile
> sdpctl profile ls
no current profile is set, run 'sdpctl profile set'

Available profiles
Name         Config directory
----         ----------------
secondary    /home/larskajes/.config/sdpctl/profiles/secondary
```

This seems very edge case, but I bet we're going to have some user run into this if not fixed :P.